roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -20,147 +20,168 @@
+      FcnMode fcnMode = FcnMode::classic; // local config
+   };
+
+   explicit RooMinimizer(RooAbsReal &function, FcnMode fcnMode = FcnMode::classic);
+   explicit RooMinimizer(RooAbsReal &function, Config &cfg);
+   explicit RooMinimizer(std::shared_ptr<RooFit::TestStatistics::RooAbsL> likelihood, Config &cfg);
+
+   ~RooMinimizer() override;
+   RooMinimizer(const RooMinimizer &min);
+
+   enum Strategy { Speed = 0, Balance = 1, Robustness = 2 };
+   enum PrintLevel { None = -1, Reduced = 0, Normal = 1, ExtraForProblem = 2, Maximum = 3 };
+
+   // Setters on _theFitter
+   static void setStrategy(int strat);
+   static void setErrorLevel(double level);
+   static void setEps(double eps);
+   static void setMaxIterations(int n);
+   static void setMaxFunctionCalls(int n);
+   static void setPrintLevel(int newLevel);
+
+   // Setters on _fcn
+   void optimizeConst(int flag);
+   void setEvalErrorWall(bool flag);
+   void setRecoverFromNaNStrength(double strength);
+   void setOffsetting(bool flag);
+   void setPrintEvalErrors(int numEvalErrors);
+   void setVerbose(bool flag = true);
+   bool setLogFile(const char *logf = nullptr);
+
+   int migrad();
+   int hesse();
+   int minos();
+   int minos(const RooArgSet &minosParamList);
+   int seek();
+   int simplex();
+   int improve();
+
+   int minimize(const char *type, const char *alg = nullptr);
+
+   RooFitResult *save(const char *name = nullptr, const char *title = nullptr);
+   RooPlot *contour(RooRealVar &var1, RooRealVar &var2, double n1 = 1.0, double n2 = 2.0, double n3 = 0.0,
+                    double n4 = 0.0, double n5 = 0.0, double n6 = 0.0, unsigned int npoints = 50);
+
+   void setProfile(bool flag = true) { _cfg.profile = flag; }
+   /// Enable or disable the logging of function evaluations to a RooDataSet.
+   /// \see RooMinimizer::getLogDataSet().
+   /// param[in] flag Boolean flag to disable or enable the functionality.
+   void setLoggingToDataSet(bool flag = true) { _loggingToDataSet = flag; }
+
+   /// If logging of function evaluations to a RooDataSet is enabled, returns a
+   /// pointer to a dataset with one row per evaluation of the RooAbsReal passed
+   /// to the minimizer. As columns, there are all floating parameters and the
+   /// values they had for that evaluation.
+   /// \see RooMinimizer::setLoggingToDataSet(bool).
+   RooDataSet *getLogDataSet() const { return _logDataSet.get(); }
+
+   static int getPrintLevel();
+
+   void setMinimizerType(std::string const &type);
+   std::string const &minimizerType() const { return _cfg.minimizerType; }
+
+   static void cleanup();
+   static RooFitResult *lastMinuitFit();
+   static RooFitResult *lastMinuitFit(const RooArgList &varList);
+
+   void saveStatus(const char *label, int status)
+   {
+      _statusHistory.push_back(std::pair<std::string, int>(label, status));
+   }
+
+   int evalCounter() const;
+   void zeroEvalCount();
+
+   ROOT::Fit::Fitter *fitter();
+   const ROOT::Fit::Fitter *fitter() const;
```

Comment:
```suggestion
   static ROOT::Fit::Fitter *fitter();
```
Also this can be `static`, and then you also don't need to have a separate version for the case where the `RooMinimizer` is `const`.